tests/e2e/envfuncsext/backup.go

```diff
@@ -0,0 +1,352 @@
+package envfuncsext
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"cloud.google.com/go/storage"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/pkg/env"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+
+	appsv1alpha1 "github.com/vesoft-inc/nebula-operator/apis/apps/v1alpha1"
+	appspkg "github.com/vesoft-inc/nebula-operator/pkg/kube"
+	e2econfig "github.com/vesoft-inc/nebula-operator/tests/e2e/config"
+)
+
+type (
+	NebulaBackupInstallOptions struct {
+		Name      string
+		Namespace string
+		Spec      appsv1alpha1.BackupSpec
+	}
+
+	nebulaBackupCtxKey struct {
+		backupType string
+	}
+
+	NebulaBackupCtxValue struct {
+		Name            string
+		Namespace       string
+		BackupFileName  string
+		StorageType     string
+		BucketName      string
+		Region          string
+		CleanBackupData bool
+	}
+
+	NebulaBackupOption  func(*NebulaBackupOptions)
+	NebulaBackupOptions struct {
+		WaitOptions []wait.Option
+	}
+)
+
+func DeployNebulaBackup(incremental bool, nbCtx NebulaBackupInstallOptions) env.Func {
```

Comment:
I think check the backup state is enough, cloud vendor codes should not be introduced in e2e framework.  The reason is that you need run restore case to check the data integrity.

Reply:
Got it, will add test cases for the above.

Reply:
Added